lib/puppet/util/windows/file.rb

```diff
@@ -89,10 +89,31 @@ def symlink(target, symlink)
   end
   module_function :symlink
 
+
+  def file_exist(file_name)
+    path = file_name.to_s
```

Comment:
Note that you also lost the check for `manages_symlinks?` / the fallback to built-in Ruby behavior - which is probably OK in this case, as that simply checks for the existence of `CreateSymbolicLinkW` (not that you actually have the token privilege to use it).

I'm quite certain the end result will be the same, because `GetFileAttributes` should behave OK everywhere, it's just that `FILE_ATTRIBUTE_REPARSE_POINT` will never be returned where there's no symlink support.

👍